lib/aquifer.api.js

```diff
@@ -74,18 +78,44 @@ class Aquifer {
 
       // Traverse cwd, and find the aquifer project dir. If one does not exist
       // in the current path, the framework is not initialized.
+      this.getProjectDir()
+      .then(() => {
+        if (this.projectDir) {
+          this.initialized = true;
+        }
+        return this.getConfig();
+      })
+      .then(() => {
+        return this.switchVersion();
+      })
+      .then(() => {
+        if (this.initialized) {
+          this.project = new this.api.project(this);
+        }
+        resolve();
+      })
+      .catch((reason) => {
+        reject(reason);
+      })
+    });
+  }
+
+  /**
+   * Find the current Aquifer project root.
+   * @returns {object} promise object.
+   */
+  getProjectDir() {
+    return new Promise((resolve, reject) => {
+      // Traverse cwd, and find the aquifer project dir.
       let dir = this.cwd;
```

Comment:
@thebruce that is a good idea, but in the interest of getting this PR merged I'd prefer to handle that as a separate enhancement. This was actually largely copied from what we were already doing to find the project directory before this PR so it is in that sense unrelated to this new functionality. Don't mean to discount the value of your suggestion at all, though.